ckanext/iati_generator/actions/iati.py

```diff
@@ -342,30 +342,97 @@ def iati_file_list(context, data_dict=None):
 def iati_resources_list(context, data_dict=None):
     """
     Return a list of resources with IATIFile records, including dataset info.
-    This fn returns ALL resources with IATIFile, no pagination.
+
+    Returns items with the following structure:
+      {
+        "namespace": ...,
+        "resource": {...},
+        "dataset": {...},
+        "iati_file": {
+            "file_type": ...,
+            "is_valid": ...,
+            "last_processed_success": ...,
+            "last_error": ...,
+        },
+      }
+
+    The admin blueprint then flattens this for the template.
     """
     data_dict = data_dict or {}
     toolkit.check_access("iati_file_list", context, data_dict)
 
-    files_by_resource = h.iati_files_by_resource()
+    # Índice: (resource_id, namespace, file_type_int) -> IATIFile
+    iati_index = h.build_iati_index()
```

Comment:
Previous version of `iati_resources_list` was better
We can safely undo all changes here
If there are any specific required change, we can analize it.
This is a mayor change